tests/utils.test.ts

```diff
@@ -302,43 +295,87 @@ describe('bin/utils', () => {
-          value,
-        );
-      }),
-      { numRuns: 100 }, // Number of times to run the test
-    );
-  });
-  test('encodeEscapedReplacer should encode all escapable characters', () => {
-    fc.assert(
-      fc.property(
-        stringWithNonPrintableCharsArb,
-        stringWithNonPrintableCharsArb,
-        (key, value) => {
-          const encodedKey = binUtils.encodeEscaped(key);
-          const encodedValue = binUtils.encodeEscaped(value);
-          const object = {
-            [key]: value,
-            [key]: {
-              [key]: value,
-            },
-            [key]: [value],
-          };
-          const encodedObject = {
-            [encodedKey]: encodedValue,
-            [encodedKey]: {
-              [encodedKey]: encodedValue,
-            },
-            [encodedKey]: [encodedValue],
-          };
-          const output = JSON.stringify(object, binUtils.encodeEscapedReplacer);
-          expect(JSON.parse(output)).toEqual(encodedObject);
+  test.prop([stringWithNonPrintableCharsArb], { numRuns: 100 })(
+    'encodeEscaped should encode all escapable characters',
+    (value) => {
+      expect(binUtils.decodeEscaped(binUtils.encodeEscaped(value))).toBe(value);
+    },
+  );
+  test.prop([stringWithNonPrintableCharsArb, stringWithNonPrintableCharsArb], {
+    numRuns: 100,
+  })(
+    'encodeEscapedReplacer should encode all escapable characters',
+    (key, value) => {
+      const encodedKey = binUtils.encodeEscaped(key);
+      const encodedValue = binUtils.encodeEscaped(value);
+      const object = {
+        [key]: value,
+        [key]: {
+          [key]: value,
         },
-      ),
-      { numRuns: 100 }, // Number of times to run the test
-    );
-  });
+        [key]: [value],
+      };
+      const encodedObject = {
+        [encodedKey]: encodedValue,
+        [encodedKey]: {
+          [encodedKey]: encodedValue,
+        },
+        [encodedKey]: [encodedValue],
+      };
+      const output = JSON.stringify(object, binUtils.encodeEscapedReplacer);
+      expect(JSON.parse(output)).toEqual(encodedObject);
+    },
+  );
+});
+
+describe('parsers', () => {
+  const vaultNameArb = fc.stringOf(
+    fc.char().filter((c) => binParsers.vaultNameRegex.test(c)),
+    { minLength: 1, maxLength: 100 },
+  );
+  const singleSecretPathArb = fc.stringOf(
+    fc.char().filter((c) => binParsers.secretPathRegex.test(c)),
+    { minLength: 1, maxLength: 25 },
+  );
+  const secretPathArb = fc
+    .array(singleSecretPathArb, { minLength: 1, maxLength: 5 })
+    .map((segments) => path.join(...segments));
+  const valueFirstCharArb = fc.char().filter((c) => /^[a-zA-Z_]$/.test(c));
+  const valueRestCharArb = fc.stringOf(
+    fc.char().filter((c) => /^[\w]$/.test(c)),
+    { minLength: 1, maxLength: 100 },
+  );
+  const valueDataArb = fc
+    .tuple(valueFirstCharArb, valueRestCharArb)
+    .map((components) => components.join(''));
```

Comment:
This probably isn't the best way to generate testing data, but this is the only way I found to ensure the tested data will be valid at all times. Instead of using the actual regex, I might need to use some other arb rule, but it would become pretty complex pretty fast.

Reply:
Regex is fine. What's wrong with regex?